Cargo.toml

```diff
@@ -687,6 +687,16 @@ description = "Demonstrates the creation and registration of a custom plugin"
 category = "Application"
 wasm = true
 
+[[example]]
```

Comment:
This shouldn't be an example. The typical way people should be using multiple apps is with sub apps.

Reply:
This is the code to reproduce the deadlock.

Reply:
It's probably better to have this as a unit test of some kind to ensure CI catches it failing in the case of a deadlock, and examples are user facing. They're expected to show users how to do something, not validate that something is working as intended.

Reply:
I agree all of you, but kinda not bother to do this until this confirms is the fix.

Reply:
@james7132 unit tests are running parallel and not running in main thread I guess. Also the thread local is not cleaned between runs, is there some integration test that runs as a stand alone process?

Reply:
The tests in `tests/` at the top level are I think mostly dedicated to integration tests, could put it there.

Reply:
> Also the thread local is not cleaned between runs

does this matter? I would think the test would still deadlock in the old code and run fine with this pr.

Reply:
> > Also the thread local is not cleaned between runs
> 
> does this matter? I would think the test would still deadlock in the old code and run fine with this pr.

It matters in some degree, the issue itself is causing main thread block, which normal unit test not running on. Also it is not 100% reproduce in each update cycle.. 
Also this case is covered by @NiklasEi 's integration test, so i prefer not introduce a new test case here. Make sense? @hymm @james7132

Reply:
The integration tests I was writing about are in a third party plugin. I think it would make sense to have an integration test in Bevy that prevents any regressions on this bug.
I think it should be possible to just convert the example you had in an earlier commit into an integration test.

Reply:
@NiklasEi I see. After some investigation, it seems existing ci workflow "run-examples" is a perfect place to check this. There is an expected frame_count configuration for each example, it only reports succeed when the frame_count met. Also,"load_gltf" example is already enabled, just missing the "debug_asset_server" feature. I'll open a new pr for this.